packages/promise-kit/index.js

```diff
@@ -0,0 +1,89 @@
+/* global globalThis */
+// @ts-check
+
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+/** @type {PromiseConstructor} */
+const BestPipelinablePromise = globalThis.HandledPromise || Promise;
+
+/**
+ * @template T
+ * @typedef {Object} PromiseKit A reified Promise
+ * @property {(value: ERef<T>) => void} resolve
+ * @property {(reason: any) => void} reject
+ * @property {Promise<T>} promise
+ */
+
+/**
+ * PromiseRecord is deprecated in favor of PromiseKit.
+ *
+ * @template T
+ * @typedef {PromiseKit<T>} PromiseRecord
+ */
+
+/**
+ * @template T
+ * @typedef {T | PromiseLike<T>} ERef
+ * A reference of some kind for to an object of type T. It may be a direct
+ * reference to a local T. It may be a local presence for a remote T. It may
+ * be a promise for a local or remote T. Or it may even be a thenable
+ * (a promise-like non-promise with a "then" method) for a T.
+ */
+
+/**
+ * Needed to prevent type errors where functions are detected to be undefined.
+ */
+const NOOP_INITIALIZER = harden(() => {});
+
+/**
+ * makePromiseKit() builds a Promise object, and returns a record
+ * containing the promise itself, as well as separate facets for resolving
+ * and rejecting it.
+ *
+ * @template T
+ * @returns {PromiseKit<T>}
+ */
+export function makePromiseKit() {
+  /** @type {(value: ERef<T>) => void} */
+  let res = NOOP_INITIALIZER;
+  /** @type {(reason: any) => void} */
+  let rej = NOOP_INITIALIZER;
+
+  /** @type {Promise<any> & {domain?: unknown}} */
```

Comment:
```suggestion
  /** @type {Promise<any>} */
```
or perhaps this simpler type doesn't need to be present at all.

Reply:
I believe the ` & {domain?: unknown}` wart was to allow accessing that property further down.